src/widgets/discardAllChanges.ts

```diff
@@ -0,0 +1,40 @@
+import { showDialog, Dialog, showErrorMessage } from '@jupyterlab/apputils';
+import { TranslationBundle } from '@jupyterlab/translation';
+import { IGitExtension } from '../tokens';
+
+/**
+ * Discard changes in all unstaged and staged files
+ *
+ * @param isFallback If dialog is called when the classical pull operation fails
+ */
+export async function discardAllChanges(
+  model: IGitExtension,
+  trans: TranslationBundle,
+  isFallback?: boolean
+): Promise<void> {
+  const result = await showDialog({
+    title: trans.__('Discard all changes'),
+    body: isFallback
+      ? trans.__(
+          'Do you want to permanently discard changes to all files and pull? This action cannot be undone.'
```

Comment:
I rephrase a bit the message

```suggestion
          'Your current changes forbid pulling the latest changes. Do you want to permanently discard those changes? This action cannot be undone.'
```